src/infrasys/system.py

```diff
@@ -284,6 +285,65 @@ def from_dict(
         logger.info("Deserialized system {}", system.label)
         return system
 
+    def save(
+        self,
+        fpath: Path | str,
+        filename: str = "system.json",
+        zip: bool = False,
+        overwrite: bool = False,
+    ) -> None:
+        """Save the contents of a system and the Time series in a single directory.
+
+        By default, this method creates the user specified folder using the
+        `to_json` method. If user sets `zip = True`, we create the folder of
+        the user (if it does not exists), zip it to the same location specified
+        and delete the folder.
+
+        Parameters
+        ----------
+        fpath : Path | str
+           Filepath to write the contents of the system.
+        zip : bool
+            Set to True if you want to archive to a zip file.
+        filename: str
+            Name of the sytem to serialize. Default value: "system.json".
+        overwrite: bool
+            Overwrites the system if it already exist on the fpath.
+
+        Examples
+        --------
+        >>> system.save("/path/of/your/choice/my_system")
+        INFO: Wrote system data to systems/system.json
+        INFO: Copied time series data to systems/my_system/system_time_series
+
+        >>> system.save("/home/$USER/systems/my_system", zip=True)
+        INFO: Wrote system data to systems/system1.json
```

Comment:
These names don't line up. Also, do we have a print showing the zip filename?

Reply:
I updated the doctoring so that it matches up and added the logging for the zip file.